frontend/lib/strapi-sdk/zod.ts

```diff
@@ -0,0 +1,190 @@
+import { z } from 'zod';
+import {
+   BooleanFilterInput,
+   DateTimeFilterInput,
+   IdFilterInput,
+   IntFilterInput,
+   ProductListFiltersInput,
+   StringFilterInput,
+} from './generated/sdk';
+
+type Properties<T> = Required<{
+   [K in keyof T]: z.ZodType<T[K], any, T[K]>;
+}>;
+
+export function BooleanFilterInputSchema(): z.ZodObject<
+   Properties<BooleanFilterInput>
+> {
+   return z.object<Properties<BooleanFilterInput>>({
+      and: z.array(z.boolean().nullable()).nullish(),
+      between: z.array(z.boolean().nullable()).nullish(),
+      contains: z.boolean().nullish(),
+      containsi: z.boolean().nullish(),
+      endsWith: z.boolean().nullish(),
+      eq: z.boolean().nullish(),
+      eqi: z.boolean().nullish(),
+      gt: z.boolean().nullish(),
+      gte: z.boolean().nullish(),
+      in: z.array(z.boolean().nullable()).nullish(),
+      lt: z.boolean().nullish(),
+      lte: z.boolean().nullish(),
+      ne: z.boolean().nullish(),
+      not: z.lazy(() => BooleanFilterInputSchema().nullish()),
+      notContains: z.boolean().nullish(),
+      notContainsi: z.boolean().nullish(),
+      notIn: z.array(z.boolean().nullable()).nullish(),
+      notNull: z.boolean().nullish(),
+      null: z.boolean().nullish(),
+      or: z.array(z.boolean().nullable()).nullish(),
+      startsWith: z.boolean().nullish(),
```

Comment:
Seems like this section is repeated 3 or 4 times and could easily be extracted to a variable / constant.